service/frontend/workflowHandler.go

```diff
@@ -1002,8 +1003,12 @@ func (wh *WorkflowHandler) RespondDecisionTaskFailed(ctx context.Context, reques
 		wh.GetThrottledLogger(),
 		tag.BlobSizeViolationOperation("RespondDecisionTaskFailed"),
 	); err != nil {
+		serverFailure := failure.NewServerFailure(common.FailureReasonFailureExceedsLimit, false)
+		serverFailure.Cause = failure.Truncate(request.Failure, sizeLimitWarn)
+		request.Failure = serverFailure
+
 		// details exceed, we would just truncate the size for decision task failed as the details is not used anywhere by client code
-		request.Details = nil
+		request.Failure = nil
```

Comment:
I'm confused by this piece of logic.  First we set `request.Failure` to `serverFailure` and then we immediately set it to nil.

Reply:
I literally forgot to remove these 3 lines after I wrote 3 lines above. Great catch!